### PR TITLE
feat(lint): add noProcessEnv

### PR DESCRIPTION
**Pull request by unvalley**
## Summary

Implement a [no-porcess-env](https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-process-env.md) lint rule from eslint-plugin-n: 

ref: https://github.com/biomejs/biome/discussions/3#discussioncomment-10158971

## Test Plan

Snapshot

### CODE REVIEW

**Comment by ematipico** [comment]
You might want to use the semantic model, and not trigger the rule if `process` isn't a global variable

**Comment by Conaclos** [comment]
> You might want to use the semantic model, and not trigger the rule if process isn't a global variable

Note that `process` can be imported:

```js
import * as process from "node:process";
import process from "node:process";
```

File: crates/biome_js_analyze/src/lint/nursery/no_process_env.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Don't use "<Emphasis>"process.env"</Emphasis>"."
+                },
+            )
+            .note(markup! {
+                "The use of "<Emphasis>"process.env"</Emphasis>" is discouraged; use a centralized configuration file instead for better maintainability and deployment consistency."
```
Comment (Conaclos) [inline crates/biome_js_analyze/src/lint/nursery/no_process_env.rs:73]: I think that we could omit the first sentence.

```suggestion
                "Use a centralized configuration file instead for better maintainability and deployment consistency."
```

**Review by Conaclos — APPROVED** [review]

**Comment by unvalley** [comment]
I've updated to use semantic model and global identifier. This is almost the same as `no_console_log`, and for now, it doesn't considering cases where it is imported.

**Pull request merged**